### PR TITLE
[WIP]Ports  integ_column_stability to C++/Kokkos

### PR DESCRIPTION
**Pull request by singhbalwinder**
Property test for "integ_column_stabilibity" has already been implemented therefore this PR implements just the fortran->C++ bridge. BFB unit test has also been added to this PR.

### CODE REVIEW

File: components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  static constexpr auto troppres = 80000;
+
+  //Use parallel_reduce to compute reduction in brunt_int
+  Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, nlev_pack), [&] (const Int& k, Scalar& val) {
+      //Find if pressure is greater than tropospheric pressure
+      auto press_gt_troppress = (pres(k) > troppres);
+
+      Spack my_result(0);
+      my_result.set(press_gt_troppress, dz_zt(k) * brunt(k));
+      for (int s = 0; s < Spack::n; ++s) {
```
Comment (ambrad) [inline components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp:31]: Nice. Would you add a TODO comment here noting that we should add `sum(const Pack&)` to `ekat_pack.hpp`? Once that's available, this loop can be replaced with `val += sum(my_result)`.

Comment (jeff-cohere) [reply components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp:31]: I'll add an issue for this to the ekat repo.

File: components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const MemberType& team,
+  const Int& nlev,
+  const uview_1d<const Spack>& dz_zt,
+  const uview_1d<const Spack>& pres,
+  const uview_1d<const Spack>& brunt,
+  Scalar& brunt_int)
+{
+  const Int nlev_pack = ekat::pack::npack<Spack>(nlev);
+  brunt_int = 0;
+  static constexpr auto troppres = 80000;
```
Comment (ambrad) [inline components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp:22]: This should go into a SHOC parameters struct corresponding to the shoc.F90 comment `! Below define some parameters for SHOC`. It also needs Scalar on the LHS for BFB handling of single precision.

File: components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp (hunk 1 above)
Comment (singhbalwinder) [reply components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp:31]: I will add a TODO comment. Thanks @jeff-cohere for creating the issue.
I was also expecting something like "sum" or a one line command, so I was struggling with this implementation. @jgfouca helped with this logic.

File: components/scream/src/physics/shoc/shoc_integ_column_stability.cpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include "shoc_integ_column_stability_impl.hpp"
+#include "share/scream_types.hpp"
+
+namespace scream {
+namespace shoc {
+
+/*
+ * Explicit instantiation for computing brunt_int on Reals using the
```
[the remark this answers is not in the document]
Comment (singhbalwinder) [reply components/scream/src/physics/shoc/shoc_integ_column_stability.cpp:8]: good point. I saw this in one other file. I will fix that as well.

File: components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp (hunk 2 above)
Comment (singhbalwinder) [reply components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp:22]: Thanks. I will replace `auto` with `Scalar`. I thought about adding it to the parameter list. It is not being used at any other place in SHOC therefore, I thought about keeping it close to the only place where it is being used. I will move it to SHOC parameters file if that is more consistent or desirable.

**Comment by singhbalwinder** [comment]
Replaced by #755

**Pull request closed without merging**